lib/driver.js

```diff
@@ -100,6 +100,7 @@ class XCUITestDriver extends BaseDriver {
       let { device, udid, realDevice } = await this.determineDevice();
       log.info(`Determining device to run tests on: udid: '${udid}', real device: ${realDevice}`);
       this.device = device;
+      this.opts.device = device;
       this.opts.udid = udid;
       this.opts.isRealDevice = realDevice;
```

Comment:
Lets decide on whether we want to put everything under `this.opts.<>` or just property.
Do remember that we have to pass this down to webdriveragent.js. Its easy to pass `opts`. Otherwise constructor arguments will explode at one point.